olive/passes/pytorch/torch_trt_conversion.py

```diff
@@ -0,0 +1,172 @@
+            ),
+            "layer_name_filter": PassConfigParam(
+                type_=Union[str, List[str]],
+                default_value=None,
+                description="Only convert layers whose name contains the given string(s).",
+            ),
+            "float16": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Convert entire model to fp16. If False, only the sparse modules are converted to fp16.",
+            ),
+        }
+
+    def validate_search_point(
+        self, search_point: Dict[str, Any], accelerator_spec: AcceleratorSpec, with_fixed_value: bool = False
+    ) -> bool:
+        if accelerator_spec.accelerator_type != Device.GPU:
+            logger.info("TorchTRTConversion only supports GPU.")
+            return False
+        return True
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModel, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> PyTorchModel:
+        from olive.passes.pytorch.trt_utils import compile_trt_model
+
+        model_config = model.get_model_config()
+        model_type = model_config.model_type
+        if model_type not in seqlens:
+            raise ValueError(f"Unsupported model type: {model_type}. Supported types: {seqlens.keys()}")
+
+        if not torch.cuda.is_available():
+            raise ValueError("TorchTRTConversion requires a GPU to run.")
+        device = "cuda"
+
+        # load_data
+        assert config["data_config"] is not None, "Data config is required for TorchTRTConversion."
+        first_batch = self._data_config.to_data_container().get_first_batch(data_root_path=data_root)[0]
+        first_batch = tensor_data_to_device(first_batch, device=device)
+        batch_size = first_batch["input_ids"].shape[0]
+        seqlen = seqlens[model_type]
+
+        # load model
+        pytorch_model = model.load_model()
+        # we will update the model inplace
+        # since the models are large, it is expensive to copy and maintain two copies
+        # set model.model to None so that the input model doesn't use this same model object when it is loaded
+        model.model = None
+        # alternative is to copy the model and use the copy
+        # pytorch_model = copy.deepcopy(model.model)
+        pytorch_model.eval()
+        # move model to device
+        pytorch_model.to(device=device)
+        # convert model to fp16 if needed
+        if config["float16"]:
+            pytorch_model = pytorch_model.to(dtype=torch.float16)
+        # disable cache
+        use_cache = pytorch_model.config.use_cache
+        pytorch_model.config.use_cache = False
+
+        # get module list of layers
+        layers = get_layers(pytorch_model, model_type)
+
+        # get layer information
+        min_layer = config["min_layer"] or 0
+        max_layer = config["max_layer"] or len(layers)
```

Comment:
do we need verify whether min_layer and max_layer is valid or not? what if user specify max_layer greater than len(layers)?

Reply:
Oh yes, I was thinking of the same but didn’t get around to adding this check. Also needed for the sparsegpt pass 

I will add it. Thanks!

Reply:
Done!